File: MetodosOrdenamiento/selection/selection.cpp
#include <iostream>
#include <vector>
#include <random>
#include <algorithm>
#include <ctime>

using namespace std;

void selectionSort(vector<int>& arr) {
    int n = arr.size();
    
    for (int i = 0; i < n; i++) {
        int min_idx = i;
        for (int j = i + 1; j < n; j++) {
            if (arr[j] < arr[min_idx]) {
                min_idx = j;
            }
        }
        swap(arr[i], arr[min_idx]);
    }
}

int main() {
    srand(time(0));
    
    int tamaño;
    while (true) {
        cout << "Ingrese el tamaño del arreglo: ";
        cin >> tamaño;
        
        if (cin.fail() || tamaño <= 0) {
            cin.clear();
            cin.ignore(1000, '\n');
            cout << "Por favor, ingrese un número positivo mayor que 0." << endl;
        } else {
            break;
        }
    }
    
    vector<int> arreglo;
    for (int i = 0; i < tamaño; i++) {
        arreglo.push_back(rand() % 100 + 1);
    }
    
    cout << "\nArreglo original: ";
    for (int num : arreglo) {
        cout << num << " ";
    }
    cout << endl;
    
    vector<int> arreglo_ordenado = arreglo;
    selectionSort(arreglo_ordenado);
    
    cout << "Arreglo ordenado: ";
    for (int num : arreglo_ordenado) {
        cout << num << " ";
    }
    cout << endl;
    
    return 0;
}

File: MetodosOrdenamiento/selection/selection.py
import random

def selection_sort(arr):
    n = len(arr)
    
    # Recorremos todo el arreglo
    for i in range(n):
        # Encontramos el índice del elemento mínimo en la parte no ordenada
        min_idx = i
        for j in range(i + 1, n):
            if arr[j] < arr[min_idx]:
                min_idx = j
        
        # Intercambiamos el elemento mínimo encontrado con el primer elemento
        arr[i], arr[min_idx] = arr[min_idx], arr[i]
    
    return arr

def main():
    while True:
        try:
            tamaño = int(input("Ingrese el tamaño del arreglo: "))
            if tamaño > 0:
                break
            else:
                print("Por favor, ingrese un número positivo mayor que 0.")
        except ValueError:
            print("Por favor, ingrese un número válido.")
    
    arreglo = [random.randint(1, 100) for _ in range(tamaño)]
    
    print(f"\nArreglo original: {arreglo}")
    
    arreglo_ordenado = selection_sort(arreglo.copy())
    
    print(f"Arreglo ordenado: {arreglo_ordenado}")

if __name__ == "__main__":
    main()